### ISSUE TITLE
"docker: Error response from daemon: OCI runtime create failed"

### ISSUE THREAD

**Issue by miguelmota** [kata-containers/runtime#646]
# Description of problem

Getting error ` invalid value of -m option maxmem: maximum memory size (0x7d600000) must be at least the initial memory size (0x80000000)` when following [kata for docker on fedora tutorial](https://github.com/kata-containers/documentation/blob/master/install/docker/fedora-docker-install.md)

# Expected result

no failure

# Actual result

```bash
[fedora@ip-172-30-0-179 ~]$ sudo docker run busybox uname -a
docker: Error response from daemon: OCI runtime create failed: qemu-lite-system-x86_64: -m 2048M,slots=2,maxmem=2006M: invalid value of -m option maxmem: maximum memory size (0x7d600000) must be at least the initial memory size (0x80000000): unknown.
```

---

[fedora@ip-172-30-0-179 ~]$ sudo kata-collect-data.sh

# Meta details

Running `kata-collect-data.sh` version `1.2.0 (commit 0bcb32f)` at `2018-08-24.21:29:29.567206188+0000`.

---

Runtime is `/usr/bin/kata-runtime`.

# `kata-env`

Output of "`/usr/bin/kata-runtime kata-env`":
```
[Meta]
  Version = "1.0.13"

[Runtime]
  Debug = false
  [Runtime.Version]
    Semver = "1.2.0"
    Commit = "0bcb32f"
    OCI = "1.0.1"
  [Runtime.Config]
    Path = "/usr/share/defaults/kata-containers/configuration.toml"

[Hypervisor]
  MachineType = "pc"
  Version = "QEMU emulator version 2.11.0\nCopyright (c) 2003-2017 Fabrice Bellard and the QEMU Project developers"
  Path = "/usr/bin/qemu-lite-system-x86_64"
  BlockDeviceDriver = "virtio-scsi"
  Msize9p = 8192
  Debug = false
  UseVSock = false

[Image]
  Path = "/usr/share/kata-containers/kata-containers-image_clearlinux_1.2.0_agent_fcfa054a757.img"

[Kernel]
  Path = "/usr/share/kata-containers/vmlinuz-4.14.51.7-134.1.container"
  Parameters = ""

[Initrd]
  Path = ""

[Proxy]
  Type = "kataProxy"
  Version = "kata-proxy version 1.2.0-1796218"
  Path = "/usr/libexec/kata-containers/kata-proxy"
  Debug = false

[Shim]
  Type = "kataShim"
  Version = "kata-shim version 1.2.0-0a37760"
  Path = "/usr/libexec/kata-containers/kata-shim"
  Debug = false

[Agent]
  Type = "kata"

[Host]
  Kernel = "4.16.16-300.fc28.x86_64"
  Architecture = "amd64"
  VMContainerCapable = true
  [Host.Distro]
    Name = "Fedora"
    Version = "28"
  [Host.CPU]
    Vendor = "GenuineIntel"
    Model = "Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz"
```

---

# Runtime config files

## Runtime default config files

```
/etc/kata-containers/configuration.toml
/usr/share/defaults/kata-containers/configuration.toml
```

## Runtime config file contents

Config file `/etc/kata-containers/configuration.toml` not found
Output of "`cat "/usr/share/defaults/kata-containers/configuration.toml"`":
```
# Copyright (c) 2017-2018 Intel Corporation
#
# SPDX-License-Identifier: Apache-2.0
#

# XXX: WARNING: this file is auto-generated.
# XXX:
# XXX: Source file: "cli/config/configuration.toml.in"
# XXX: Project:
# XXX:   Name: Kata Containers
# XXX:   Type: kata

[hypervisor.qemu]
path = "/usr/bin/qemu-lite-system-x86_64"
kernel = "/usr/share/kata-containers/vmlinuz.container"
image = "/usr/share/kata-containers/kata-containers.img"
machine_type = "pc"

# Optional space-separated list of options to pass to the guest kernel.
# For example, use `kernel_params = "vsyscall=emulate"` if you are having
# trouble running pre-2.15 glibc.
#
# WARNING: - any parameter specified here will take priority over the default
# parameter value of the same name used to start the virtual machine.
# Do not set values here unless you understand the impact of doing so as you
# may stop the virtual machine from booting.
# To see the list of default parameters, enable hypervisor debug, create a
# container and look for 'default-kernel-parameters' log entries.
kernel_params = ""

# Path to the firmware.
# If you want that qemu uses the default firmware leave this option empty
firmware = ""

# Machine accelerators
# comma-separated list of machine accelerators to pass to the hypervisor.
# For example, `machine_accelerators = "nosmm,nosmbus,nosata,nopit,static-prt,nofw"`
machine_accelerators=""

# Default number of vCPUs per SB/VM:
# unspecified or 0                --> will be set to 1
# < 0                             --> will be set to the actual number of physical cores
# > 0 <= number of physical cores --> will be set to the specified number
# > number of physical cores      --> will be set to the actual number of physical cores
default_vcpus = 1

# Default maximum number of vCPUs per SB/VM:
# unspecified or == 0             --> will be set to the actual number of physical cores or to the maximum number
#                                     of vCPUs supported by KVM if that number is exceeded
# > 0 <= number of physical cores --> will be set to the specified number
# > number of physical cores      --> will be set to the actual number of physical cores or to the maximum number
#                                     of vCPUs supported by KVM if that number is exceeded
# WARNING: Depending of the architecture, the maximum number of vCPUs supported by KVM is used when
# the actual number of physical cores is greater than it.
# WARNING: Be aware that this value impacts the virtual machine's memory footprint and CPU
# the hotplug functionality. For example, `default_maxvcpus = 240` specifies that until 240 vCPUs
# can be added to a SB/VM, but the memory footprint will be big. Another example, with
# `default_maxvcpus = 8` the memory footprint will be small, but 8 will be the maximum number of
# vCPUs supported by the SB/VM. In general, we recommend that you do not edit this variable,
# unless you know what are you doing.
default_maxvcpus = 0

# Bridges can be used to hot plug devices.
# Limitations:
# * Currently only pci bridges are supported
# * Until 30 devices per bridge can be hot plugged.
# * Until 5 PCI bridges can be cold plugged per VM.
#   This limitation could be a bug in qemu or in the kernel
# Default number of bridges per SB/VM:
# unspecified or 0   --> will be set to 1
# > 1 <= 5           --> will be set to the specified number
# > 5                --> will be set to 5
default_bridges = 1

# Default memory size in MiB for SB/VM.
# If unspecified then it will be set 2048 MiB.
#default_memory = 2048

# Disable block device from being used for a container's rootfs.
# In case of a storage driver like devicemapper where a container's
# root file system is backed by a block device, the block device is passed
# directly to the hypervisor for performance reasons.
# This flag prevents the block device from being passed to the hypervisor,
# 9pfs is used instead to pass the rootfs.
disable_block_device_use = false

# Block storage driver to be used for the hypervisor in case the container
# rootfs is backed by a block device. This is either virtio-scsi or
# virtio-blk.
block_device_driver = "virtio-scsi"

# Enable iothreads (data-plane) to be used. This causes IO to be
# handled in a separate IO thread. This is currently only implemented
# for SCSI.
#
enable_iothreads = false

# Enable pre allocation of VM RAM, default false
# Enabling this will result in lower container density
# as all of the memory will be allocated and locked
# This is useful when you want to reserve all the memory
# upfront or in the cases where you want memory latencies
# to be very predictable
# Default false
#enable_mem_prealloc = true

# Enable huge pages for VM RAM, default false
# Enabling this will result in the VM memory
# being allocated using huge pages.
# This is useful when you want to use vhost-user network
# stacks within the container. This will automatically
# result in memory pre allocation
#enable_hugepages = true

# Enable swap of vm memory. Default false.
# The behaviour is undefined if mem_prealloc is also set to true
#enable_swap = true

# This option changes the default hypervisor and kernel parameters
# to enable debug output where available. This extra output is added
# to the proxy logs, but only when proxy debug is also enabled.
#
# Default false
#enable_debug = true

# Disable the customizations done in the runtime when it detects
# that it is running on top a VMM. This will result in the runtime
# behaving as it would when running on bare metal.
#
#disable_nesting_checks = true

# This is the msize used for 9p shares. It is the number of bytes
# used for 9p packet payload.
#msize_9p = 8192

# If true and vsocks are supported, use vsocks to communicate directly
# with the agent and no proxy is started, otherwise use unix
# sockets and start a proxy to communicate with the agent.
# Default false
#use_vsock = true

[factory]
# VM templating support. Once enabled, new VMs are created from template
# using vm cloning. They will share the same initial kernel, initramfs and
# agent memory by mapping it readonly. It helps speeding up new container
# creation and saves a lot of memory if there are many kata containers running
# on the same host.
#
# When disabled, new VMs are created from scratch.
#
# Default false
#enable_template = true

[proxy.kata]
path = "/usr/libexec/kata-containers/kata-proxy"

# If enabled, proxy messages will be sent to the system log
# (default: disabled)
#enable_debug = true

[shim.kata]
path = "/usr/libexec/kata-containers/kata-shim"

# If enabled, shim messages will be sent to the system log
# (default: disabled)
#enable_debug = true

[agent.kata]
# There is no field for this section. The goal is only to be able to
# specify which type of agent the user wants to use.

[runtime]
# If enabled, the runtime will log additional debug messages to the
# system log
# (default: disabled)
#enable_debug = true
#
# Internetworking model
# Determines how the VM should be connected to the
# the container network interface
# Options:
#
#   - bridged
#     Uses a linux bridge to interconnect the container interface to
#     the VM. Works for most cases except macvlan and ipvlan.
#
#   - macvtap
#     Used when the Container network interface can be bridged using
#     macvtap.
internetworking_model="macvtap"
```

---

# Image details

```
---
osbuilder:
  url: "https://github.com/kata-containers/osbuilder"
  version: "unknown"
rootfs-creation-time: "2018-08-13T22:51:39.765008919+0000Z"
description: "osbuilder rootfs"
file-format-version: "0.0.2"
architecture: "x86_64"
base-distro:
  name: "Clear"
  version: "24400"
  packages:
    default:
      - "iptables-bin"
      - "libudev0-shim"
      - "systemd"
    extra:

agent:
  url: "https://github.com/kata-containers/agent"
  name: "kata-agent"
  version: "1.2.0-fcfa054a757e7c17afba47b0b4d7e91cbb8688ed"
  agent-is-init-daemon: "no"
```

---

# Initrd details

No initrd

---

# Logfiles

## Runtime logs

Recent runtime problems found in system journal:
```
time="2018-08-24T21:27:16.959693325Z" level=warning msg="fetch sandbox device failed" arch=amd64 command=create container=e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51 error="open /run/vc/sbs/e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51/devices.json: no such file or directory" name=kata-runtime pid=1741 sandbox=e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51 sandboxid=e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51 source=virtcontainers subsystem=sandbox
time="2018-08-24T21:27:17.110223357Z" level=error msg="Unable to launch qemu: exit status 1" arch=amd64 command=create container=e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51 name=kata-runtime pid=1741 source=virtcontainers subsystem=qmp
time="2018-08-24T21:27:17.11052701Z" level=error msg="qemu-lite-system-x86_64: -m 2048M,slots=2,maxmem=2006M: invalid value of -m option maxmem: maximum memory size (0x7d600000) must be at least the initial memory size (0x80000000)\n" arch=amd64 command=create container=e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51 name=kata-runtime pid=1741 source=virtcontainers subsystem=qmp
time="2018-08-24T21:27:17.121462342Z" level=error msg="qemu-lite-system-x86_64: -m 2048M,slots=2,maxmem=2006M: invalid value of -m option maxmem: maximum memory size (0x7d600000) must be at least the initial memory size (0x80000000)\n" arch=amd64 command=create container=e03d07fb86e804a900a0384f9fc9fc1a3769d98e632979512f5e797045cfab51 name=kata-runtime pid=1741 source=runtime
time="2018-08-24T21:27:43.843690445Z" level=warning msg="fetch sandbox device failed" arch=amd64 command=create container=4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37 error="open /run/vc/sbs/4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37/devices.json: no such file or directory" name=kata-runtime pid=1806 sandbox=4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37 sandboxid=4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37 source=virtcontainers subsystem=sandbox
time="2018-08-24T21:27:43.947457764Z" level=error msg="Unable to launch qemu: exit status 1" arch=amd64 command=create container=4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37 name=kata-runtime pid=1806 source=virtcontainers subsystem=qmp
time="2018-08-24T21:27:43.947680837Z" level=error msg="qemu-lite-system-x86_64: -m 2048M,slots=2,maxmem=2006M: invalid value of -m option maxmem: maximum memory size (0x7d600000) must be at least the initial memory size (0x80000000)\n" arch=amd64 command=create container=4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37 name=kata-runtime pid=1806 source=virtcontainers subsystem=qmp
time="2018-08-24T21:27:43.958280262Z" level=error msg="qemu-lite-system-x86_64: -m 2048M,slots=2,maxmem=2006M: invalid value of -m option maxmem: maximum memory size (0x7d600000) must be at least the initial memory size (0x80000000)\n" arch=amd64 command=create container=4d1a85921586cb5bbe76edb2d2025473f03d5543ca6cb2107f57fa3b42572d37 name=kata-runtime pid=1806 source=runtime
```

## Proxy logs

No recent proxy problems found in system journal.

## Shim logs

No recent shim problems found in system journal.

---

# Container manager details

Have `docker`

## Docker

Output of "`docker version`":
```
Client:
 Version:      18.05.0-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   f150324
 Built:        Wed May  9 22:16:04 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.05.0-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   f150324
  Built:        Wed May  9 22:19:55 2018
  OS/Arch:      linux/amd64
  Experimental: true
```
Output of "`docker info`":
```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 14
Server Version: 18.05.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: kata-runtime runc
Default Runtime: kata-runtime
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 0bcb32f (expected: 4fc53a81fb7c994640722ac585fa9ca548971871)
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.16.16-300.fc28.x86_64
Operating System: Fedora 28 (Cloud Edition)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 982.4MiB
Name: ip-172-30-0-179.ec2.internal
ID: SBYX:67XR:IPVH:ZIU3:W6DU:EYCV:GPVT:HGDT:OMOY:UAOD:T7AI:5ROX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```
Output of "`systemctl show docker`":
```
Type=notify
Restart=on-failure
NotifyAccess=main
RestartUSec=100ms
TimeoutStartUSec=infinity
TimeoutStopUSec=1min 30s
RuntimeMaxUSec=infinity
WatchdogUSec=0
WatchdogTimestamp=Fri 2018-08-24 21:27:11 UTC
WatchdogTimestampMonotonic=1137090638
PermissionsStartOnly=no
RootDirectoryStartOnly=no
RemainAfterExit=no
GuessMainPID=yes
MainPID=1575
ControlPID=0
FileDescriptorStoreMax=0
NFileDescriptorStore=0
StatusErrno=0
Result=success
UID=[not set]
GID=[not set]
NRestarts=0
ExecMainStartTimestamp=Fri 2018-08-24 21:27:10 UTC
ExecMainStartTimestampMonotonic=1136844629
ExecMainExitTimestampMonotonic=0
ExecMainPID=1575
ExecMainCode=0
ExecMainStatus=0
ExecStart={ path=/usr/bin/dockerd ; argv[]=/usr/bin/dockerd ; ignore_errors=no ; start_time=[Fri 2018-08-24 21:27:10 UTC] ; stop_time=[n/a] ; pid=1575 ; code=(null) ; status=0/0 }
ExecReload={ path=/bin/kill ; argv[]=/bin/kill -s HUP $MAINPID ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
Slice=system.slice
ControlGroup=/system.slice/docker.service
MemoryCurrent=60481536
CPUUsageNSec=[not set]
TasksCurrent=18
IPIngressBytes=18446744073709551615
IPIngressPackets=18446744073709551615
IPEgressBytes=18446744073709551615
IPEgressPackets=18446744073709551615
Delegate=yes
DelegateControllers=cpu cpuacct io blkio memory devices pids
CPUAccounting=no
CPUWeight=[not set]
StartupCPUWeight=[not set]
CPUShares=[not set]
StartupCPUShares=[not set]
CPUQuotaPerSecUSec=infinity
IOAccounting=no
IOWeight=[not set]
StartupIOWeight=[not set]
BlockIOAccounting=no
BlockIOWeight=[not set]
StartupBlockIOWeight=[not set]
MemoryAccounting=yes
MemoryLow=0
MemoryHigh=infinity
MemoryMax=infinity
MemorySwapMax=infinity
MemoryLimit=infinity
DevicePolicy=auto
TasksAccounting=yes
TasksMax=infinity
IPAccounting=no
UMask=0022
LimitCPU=infinity
LimitCPUSoft=infinity
LimitFSIZE=infinity
LimitFSIZESoft=infinity
LimitDATA=infinity
LimitDATASoft=infinity
LimitSTACK=infinity
LimitSTACKSoft=8388608
LimitCORE=infinity
LimitCORESoft=infinity
LimitRSS=infinity
LimitRSSSoft=infinity
LimitNOFILE=infinity
LimitNOFILESoft=infinity
LimitAS=infinity
LimitASSoft=infinity
LimitNPROC=infinity
LimitNPROCSoft=infinity
LimitMEMLOCK=16777216
LimitMEMLOCKSoft=16777216
LimitLOCKS=infinity
LimitLOCKSSoft=infinity
LimitSIGPENDING=3816
LimitSIGPENDINGSoft=3816
LimitMSGQUEUE=819200
LimitMSGQUEUESoft=819200
LimitNICE=0
LimitNICESoft=0
LimitRTPRIO=0
LimitRTPRIOSoft=0
LimitRTTIME=infinity
LimitRTTIMESoft=infinity
OOMScoreAdjust=0
Nice=0
IOSchedulingClass=0
IOSchedulingPriority=0
CPUSchedulingPolicy=0
CPUSchedulingPriority=0
TimerSlackNSec=50000
CPUSchedulingResetOnFork=no
NonBlocking=no
StandardInput=null
StandardInputData=
StandardOutput=journal
StandardError=inherit
TTYReset=no
TTYVHangup=no
TTYVTDisallocate=no
SyslogPriority=30
SyslogLevelPrefix=yes
SyslogLevel=6
SyslogFacility=3
LogLevelMax=-1
SecureBits=0
CapabilityBoundingSet=cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_module cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend
AmbientCapabilities=
DynamicUser=no
RemoveIPC=no
MountFlags=
PrivateTmp=no
PrivateDevices=no
ProtectKernelTunables=no
ProtectKernelModules=no
ProtectControlGroups=no
PrivateNetwork=no
PrivateUsers=no
ProtectHome=no
ProtectSystem=no
SameProcessGroup=no
UtmpMode=init
IgnoreSIGPIPE=yes
NoNewPrivileges=no
SystemCallErrorNumber=0
LockPersonality=no
RuntimeDirectoryPreserve=no
RuntimeDirectoryMode=0755
StateDirectoryMode=0755
CacheDirectoryMode=0755
LogsDirectoryMode=0755
ConfigurationDirectoryMode=0755
MemoryDenyWriteExecute=no
RestrictRealtime=no
RestrictNamespaces=no
MountAPIVFS=no
KeyringMode=private
KillMode=process
KillSignal=15
SendSIGKILL=yes
SendSIGHUP=no
Id=docker.service
Names=docker.service
Requires=sysinit.target system.slice
Wants=network-online.target
Conflicts=shutdown.target
Before=shutdown.target
After=system.slice network-online.target systemd-journald.socket sysinit.target basic.target firewalld.service
Documentation=https://docs.docker.com
Description=Docker Application Container Engine
LoadState=loaded
ActiveState=active
SubState=running
FragmentPath=/usr/lib/systemd/system/docker.service
UnitFileState=disabled
UnitFilePreset=disabled
StateChangeTimestamp=Fri 2018-08-24 21:27:11 UTC
StateChangeTimestampMonotonic=1137090641
InactiveExitTimestamp=Fri 2018-08-24 21:27:10 UTC
InactiveExitTimestampMonotonic=1136844665
ActiveEnterTimestamp=Fri 2018-08-24 21:27:11 UTC
ActiveEnterTimestampMonotonic=1137090641
ActiveExitTimestamp=Fri 2018-08-24 21:27:09 UTC
ActiveExitTimestampMonotonic=1135841034
InactiveEnterTimestamp=Fri 2018-08-24 21:27:10 UTC
InactiveEnterTimestampMonotonic=1136843038
CanStart=yes
CanStop=yes
CanReload=yes
CanIsolate=no
StopWhenUnneeded=no
RefuseManualStart=no
RefuseManualStop=no
AllowIsolate=no
DefaultDependencies=yes
OnFailureJobMode=replace
IgnoreOnIsolate=no
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobRunningTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=yes
AssertResult=yes
ConditionTimestamp=Fri 2018-08-24 21:27:10 UTC
ConditionTimestampMonotonic=1136843822
AssertTimestamp=Fri 2018-08-24 21:27:10 UTC
AssertTimestampMonotonic=1136843823
Transient=no
Perpetual=no
StartLimitIntervalUSec=1min
StartLimitBurst=3
StartLimitAction=none
FailureAction=none
SuccessAction=none
InvocationID=396a3317b61646dc8ef8a5ce679bd3e4
CollectMode=inactive
```
No `kubectl`

---

# Packages

No `dpkg`
Have `rpm`
Output of "`rpm -qa|egrep "(cc-oci-runtimecc-runtimerunv|kata-proxy|kata-runtime|kata-shim|kata-containers-image|linux-container|qemu-)"`":
```
qemu-vanilla-data-2.11.2+git.a39e0b3e82-44.1.x86_64
qemu-vanilla-bin-2.11.2+git.a39e0b3e82-44.1.x86_64
kata-proxy-bin-1.2.0+git.1796218-32.1.x86_64
kata-containers-image-1.2.0-32.1.x86_64
qemu-lite-data-2.11.0+git.a39e0b3e82-47.1.x86_64
qemu-vanilla-2.11.2+git.a39e0b3e82-44.1.x86_64
kata-proxy-1.2.0+git.1796218-32.1.x86_64
kata-runtime-1.2.0+git.0bcb32f-45.1.x86_64
qemu-lite-bin-2.11.0+git.a39e0b3e82-47.1.x86_64
kata-shim-bin-1.2.0+git.0a37760-33.1.x86_64
kata-linux-container-4.14.51.7-134.1.x86_64
qemu-lite-2.11.0+git.a39e0b3e82-47.1.x86_64
kata-shim-1.2.0+git.0a37760-33.1.x86_64
```

---

**Comment by miguelmota** [comment]
Changed the `default_memory` in `/usr/share/defaults/kata-containers/configuration.toml` which seems to solve that particular memory error, but now get 

```bash
[fedora@ip-172-30-0-179 ~]$ sudo docker run busybox uname -a
docker: Error response from daemon: OCI runtime create failed: Could not access KVM kernel module: No such file or directory
qemu-lite-system-x86_64: failed to initialize KVM: No such file or directory: unknown.
```

**Comment by miguelmota** [comment]
Tried enabling the kernel module but got `Operation not supported`

```bash
[fedora@ip-172-30-0-179 ~]$ sudo modprobe kvm-intel
modprobe: ERROR: could not insert 'kvm_intel': Operation not supported
```

It looks like kata only works on bare metal? (trying it on an ec2 instance)

**Comment by caoruidong** [comment]
Do you open nest virtualization?

**Comment by miguelmota** [comment]
Not possible with EC2 (non baremetal), therefore closing issue

**Comment by jodh-intel** [comment]
I'm confused - the output of `kata-env` above suggests your environment would work. It certainly is possible to run Kata on a non-baremetal system.

To run all the hardware checks (which will tell you if the system is "Kata capable"):

```
$ sudo kata-runtime kata-check
```



See: https://github.com/kata-containers/runtime#hardware-requirements

**Comment by miguelmota** [comment]
@jodh-intel this is the output 

```bash
[fedora@ip-172-30-0-179 ~]$ sudo kata-runtime kata-check
INFO[0000] CPU property found                            arch=amd64 description="Intel Architecture CPU" name=GenuineIntel pid=9485 source=runtime type=attribute
ERRO[0000] CPU property not found                        arch=amd64 description="Virtualization support" name=vmx pid=9485 source=runtime type=flag
INFO[0000] CPU property found                            arch=amd64 description="64Bit CPU" name=lm pid=9485 source=runtime type=flag
INFO[0000] CPU property found                            arch=amd64 description=SSE4.1 name=sse4_1 pid=9485 source=runtime type=flag
INFO[0000] kernel property found                         arch=amd64 description="Kernel-based Virtual Machine" name=kvm pid=9485 source=runtime type=module
INFO[0000] kernel property found                         arch=amd64 description="Host kernel accelerator for virtio" name=vhost pid=9485 source=runtime type=module
INFO[0000] kernel property found                         arch=amd64 description="Host kernel accelerator for virtio network" name=vhost_net pid=9485 source=runtime type=module
INFO[0000] kernel property found                         arch=amd64 description="Intel KVM" name=kvm_intel pid=9485 source=runtime type=module
```

**Comment by miguelmota** [comment]
doesn't look like nested vm is supported in this fedora ec2 instance

```bash
[fedora@ip-172-30-0-179 ~]$ cat /proc/cpuinfo
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 63
model name      : Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz
stepping        : 2
microcode       : 0x3c
cpu MHz         : 2399.918
cache size      : 30720 KB
physical id     : 0
siblings        : 1
core id         : 0
cpu cores       : 1
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology cpuid pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm cpuid_fault invpcid_single pti fsgsbase bmi1 avx2 smep bmi2 erms invpcid xsaveopt
bugs            : cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass
bogomips        : 4800.10
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:
```

@jodh-intel

**Comment by ydjainopensource** [comment]
@miguelmota As the kata-check output shows, you need vmx support on the host to run KVM based virtual machines, something which kata depends on internally. 

/cc @jodh-intel However, I think runtime logs suggest something else. If not, can we work to make the log output better?

**Comment by jodh-intel** [comment]
@miguelmota - thanks for the info. However, it looks like you only pasted part of the `kata-check` output?

**Comment by miguelmota** [comment]
@jodh-intel my mistake, here's the full output

```bash
[fedora@ip-172-30-0-179 ~]$ sudo kata-runtime kata-check
INFO[0000] CPU property found                            arch=amd64 description="Intel Architecture CPU" name=GenuineIntel pid=10721 source=runtime type=attribute
ERRO[0000] CPU property not found                        arch=amd64 description="Virtualization support" name=vmx pid=10721 source=runtime type=flag
INFO[0000] CPU property found                            arch=amd64 description="64Bit CPU" name=lm pid=10721 source=runtime type=flag
INFO[0000] CPU property found                            arch=amd64 description=SSE4.1 name=sse4_1 pid=10721 source=runtime type=flag
INFO[0000] kernel property found                         arch=amd64 description="Kernel-based Virtual Machine" name=kvm pid=10721 source=runtime type=module
INFO[0000] kernel property found                         arch=amd64 description="Host kernel accelerator for virtio" name=vhost pid=10721 source=runtime type=module
INFO[0000] kernel property found                         arch=amd64 description="Host kernel accelerator for virtio network" name=vhost_net pid=10721 source=runtime type=module
INFO[0000] kernel property found                         arch=amd64 description="Intel KVM" name=kvm_intel pid=10721 source=runtime type=module
ERRO[0000] open /sys/module/kvm_intel/parameters/nested: no such file or directory  arch=amd64 name=kata-runtime pid=10721 source=runtime
open /sys/module/kvm_intel/parameters/nested: no such file or directory
[fedora@ip-172-30-0-179 ~]$
```

**Comment by jodh-intel** [comment]
@ydjainopensource - issue fixed on #660.
